File: 0443-string-compression/0443-string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        
        write = 0
        read = 0

        while read < len(chars):
            
            char = chars[read]
            count = 0

            while read < len(chars) and chars[read] == char:
                read += 1
                count += 1
            
            chars[write] = char
            write += 1

            if count > 1:
                for digit in str(count):
                    print(chars[write])
                    chars[write] = digit
                    print(chars[write])
                    print()
                    write += 1
        
        return write

File: 0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) 
    {
       
       int left = 0;
       int right = chars.size();
       int write = 0;

       while (left < right)
       {
            char curr = chars[left];
            int count = 0;

            while (left < right && curr == chars[left])
            {
                left += 1;
                count += 1;
            }

            chars[write] = curr;
            write++;

            if (count > 1)
            {
                string s = to_string(count);

                for (char digit : s)
                {
                    chars[write] = digit;
                    write++;
                }
            }
       }

       return write;
    }

private:
    int res = 0;
};